psl/builtin-connectors/src/mysql_datamodel_connector/validations.rs

```diff
@@ -89,3 +92,39 @@ pub(crate) fn field_types_can_be_used_in_a_primary_key(
         break;
     }
 }
+
+pub(crate) fn uses_native_referential_action_set_default(
+    connector: &dyn Connector,
+    field: RelationFieldWalker<'_>,
+    diagnostics: &mut Diagnostics,
+) {
+    let get_span = |referential_action_type: &str| -> Span {
+        field
+            .ast_field()
+            .span_for_argument("relation", referential_action_type)
+            .unwrap_or_else(|| field.ast_field().span())
+    };
+
+    let warning_msg = || {
+        format!(
+            "Using {set_default} on {connector} may yield to unexpected results, as the database will silently change the referential action to `{no_action}`.",
+            set_default = ReferentialAction::SetDefault.as_str(),
+            connector = connector.name(),
+            no_action = ReferentialAction::NoAction.as_str(),
+        )
+    };
+
+    if let Some(on_delete) = field.explicit_on_delete() {
+        if on_delete == ReferentialAction::SetDefault {
```

Comment:
We could instead do:

```rust
if let Some(ReferentialAction::SetDefault) = forward.explicit_on_delete()
```

Reply:
I'd recommend doing so too. Never nest too much!